### PR TITLE
CRIMAPP-310-314-366 Reorder case details attributes

### PR DESCRIPTION
**Pull request by ripan**
## Description of change
Move case related information from "Overview" section to "Case Details" section

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAPP-310
https://dsdmoj.atlassian.net/browse/CRIMAPP-314
https://dsdmoj.atlassian.net/browse/CRIMAPP-366

### Before changes:
Screenshot 2024-02-01 at 13 14 36

### After changes:
Screenshot 2024-02-01 at 13 13 27

### CODE REVIEW

File: app/views/casework/crime_applications/_case_details.html.erb [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <%= label_text(:case_details) %>
 </h2>
 <% if case_details.case_type.present? %>
-  <dl class="govuk-summary-list govuk-!-margin-bottom-9">
```
Comment (hiboabd) [inline app/views/casework/crime_applications/_case_details.html.erb:5]: Why was this removed?

File: spec/system/casework/viewing_an_application/application_details/case_details_spec.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe 'When viewing an case details' do
```
Comment (hiboabd) [inline spec/system/casework/viewing_an_application/application_details/case_details_spec.rb:3]: 'When viewing case details'

**Review by hiboabd — APPROVED** [review]
LGTM, left two non blocking comments

File: app/views/casework/crime_applications/_case_details.html.erb (hunk 1 above)
Comment (ripan) [reply app/views/casework/crime_applications/_case_details.html.erb]: Good Catch 👍 
I added 1 more <dl> tag and last <dl> should have `govuk-!-margin-bottom-9` class, but it was conditional.
Here is the conditional structure now. The new attributes should be nested under `<% if case_details.case_type.present? %>` condition
Screenshot 2024-02-02 at 12 10 47

File: app/views/casework/crime_applications/_case_details.html.erb [hunk 3]
```diff
@@ -66,77 +66,74 @@
...
-    <% if case_details.case_concluded? %>
+    <% if FeatureFlags.means_journey.enabled? %>
       <div class="govuk-summary-list__row">
         <dt class="govuk-summary-list__key">
-          <%= label_text(:date_case_concluded) %>
+          <%= label_text(:has_case_concluded) %>
         </dt>
         <dd class="govuk-summary-list__value">
-          <%= l(case_details.date_case_concluded, format: :compact) %>
+          <%= t(case_details.has_case_concluded, scope: 'values') || t('values.not_asked') %>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/_case_details.html.erb:75]: I think we're not going with this approach anymore? based on what fleur said here: https://mojdt.slack.com/archives/C02AFLHGK9S/p1706792275399049

File: app/views/casework/crime_applications/_case_details.html.erb [hunk 4]
```diff
@@ -66,77 +66,74 @@
...
-        </dd>
-      </div>
       <div class="govuk-summary-list__row">
         <dt class="govuk-summary-list__key">
-          <%= label_text(:preorder_work_details) %>
+          <%= label_text(:is_preorder_work_claimed) %>
         </dt>
         <dd class="govuk-summary-list__value">
-          <%= case_details.preorder_work_details %>
+          <%= t(case_details.is_preorder_work_claimed, scope: 'values') || t('values.not_asked') %>
```
Comment (hiboabd) [inline app/views/casework/crime_applications/_case_details.html.erb:95]: Same with this one, so it might need a conditional to hide the section if it's not asked `<% if case_details.preorder_work_claimed? %>`

**Review by hiboabd — APPROVED** [review]
👍 thanks
